src/rotator_library/providers/qwen_auth_base.py

```
@@ -524,33 +558,72 @@ async def get_api_details(self, credential_identifier: str) -> Tuple[str, str]:
+
+        URL normalization follows upstream qwen-code getCurrentEndpoint() logic:
+        - Adds https:// prefix if missing
+        - Ensures /v1 suffix is present
+        - Defaults to DashScope URL when resource_url is not set
         """
-        # Detect credential type
-        if os.path.isfile(credential_identifier):
-            # OAuth credential: file path to JSON
+
+        try:
+            is_oauth = credential_identifier.startswith("env://") or os.path.isfile(
```

Ruff (0.15.9)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)